### PR TITLE
[docs] Improve `mojo_bug_report.yaml` issue template

### LINKED ISSUE #2008: [Docs] `.github/ISSUE_TEMPLATE/mojo_bug_report.yaml` has typos
### Where is the problem?

https://github.com/modularml/mojo/blob/main/.github/ISSUE_TEMPLATE/mojo_bug_report.yaml

### What can we do better?

https://github.com/modularml/mojo/blob/main/.github/ISSUE_TEMPLATE/mojo_bug_report.yaml should have correct grammar and spelling.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by hamirmahal**
fixes #2008.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/mojo_bug_report.yaml [hunk 1]
```diff
@@ -25,7 +31,7 @@ body:
       label: Steps to reproduce
       description: Provide the specific steps to reproduce the issue.
       value: |
-        - Include relevant code snippet or link to code that did not work as expected.
+        - Include a relevant code snippet or link to code that did not work as expected.
```
Comment (VascoSch92) [inline .github/ISSUE_TEMPLATE/mojo_bug_report.yaml:34]:
```suggestion
        - Provide a relevant code snippet or a link to the code that did not work as expected.
```

Comment (JoeLoser) [reply .github/ISSUE_TEMPLATE/mojo_bug_report.yaml:34]: +1 to these suggestions.

**Review by JoeLoser — COMMENTED** [review]
Looks good once you incorporate the suggestion from @VascoSch92.

Can you please target the nightly branch instead of main as described in the [docs](https://github.com/modularml/mojo/blob/f72199c1bbd4ce7f1ed19d5911671dff31e20473/CONTRIBUTING.md#pull-requests)?  You'll also need to sign your work - see the [docs](https://github.com/modularml/mojo/blob/f72199c1bbd4ce7f1ed19d5911671dff31e20473/CONTRIBUTING.md#signing-your-work).  Thanks again for the contribution!

**Comment by hamirmahal** [comment]
You're welcome!

**Review by JoeLoser — APPROVED** [review]

**Pull request merged**